src/shared/ui/survey/RequestHealthRecordDataItem/RequestHealthRecordDataItem.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { FC, useMemo, useState } from 'react';
+import { Linking } from 'react-native';
+
+import { RequestHealthRecordDataResponse } from '@app/features/pass-survey/lib/types/payload';
+import { colors } from '@app/shared/lib/constants/colors';
+import { Box, RadioGroup, YStack } from '@app/shared/ui/base';
+import { HealthRecordIcon } from '@app/shared/ui/icons/HealthRecord';
+import { Link } from '@app/shared/ui/Link';
+import { RadioOption } from '@app/shared/ui/survey/RadioActivityItem/types';
+import { Text } from '@app/shared/ui/Text';
+
+const REQUEST_HEALTH_RECORD_DATA_LINK = 'https://mindlogger.org/';
+
+export type RequestHealthRecordDataConfig = {
+  optInOutOptions: [
+    {
+      id: 'opt_in';
+      label: string;
+    },
+    {
+      id: 'opt_out';
+      label: string;
+    },
+  ];
+};
+
+type RequestHealthRecordDataItemProps = {
+  config: RequestHealthRecordDataConfig;
+  question: string;
+  onChange: (value: RadioOption) => void;
+  initialValue?: RequestHealthRecordDataResponse;
+  textReplacer: (markdown: string) => string;
+  onValidationError?: (hasError: boolean) => void;
+};
+
+export const RequestHealthRecordDataItem: FC<
+  RequestHealthRecordDataItemProps
+> = ({
+  config,
+  question,
+  onChange,
+  initialValue,
+  textReplacer,
+  onValidationError,
+}) => {
+  const [selectedOptionId, setSelectedOptionId] = useState(
+    initialValue ?? null,
+  );
+  const questionText = useMemo(
+    () => textReplacer(question),
+    [question, textReplacer],
+  );
+
+  const options: RadioOption[] = useMemo(
+    () =>
+      config.optInOutOptions.map((option, index) => ({
+        id: option.id,
+        text: option.label,
+        image: null,
+        score: null,
+        tooltip: null,
+        color: null,
+        isHidden: false,
+        value: index,
+      })),
+    [config.optInOutOptions],
+  );
+
+  const handleValueChange = (value: string) => {
+    const selectedOptionValue = value as RequestHealthRecordDataResponse;
+    const selectedOption =
+      selectedOptionValue === 'opt_in' ? options[0] : options[1];
+
+    setSelectedOptionId(selectedOptionValue);
+
+    if (onValidationError) {
+      onValidationError(false);
+    }
+
+    onChange(selectedOption);
```

Comment:
I think two of my previous comments will be fixed by this change: 🤓 
```suggestion
    onChange(selectedOptionValue);
```

Reply:
Yup. Figured it out. Was just a simple incorrect type on the `onChange` function and value assignment 🤓